u1qns/Week_09/BOJ_1916_최소비용 구하기.java

```diff
@@ -0,0 +1,61 @@
+import java.io.*;
+import java.util.*;
+
+public class Main {
+    public static void main(String[] args) throws NumberFormatException, IOException {
+        StringTokenizer st;
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringBuilder sb = new StringBuilder();
+
+        int N = Integer.parseInt(br.readLine());
+        int M = Integer.parseInt(br.readLine());
+        int dp[] = new int[N+1];
+        int tmp[] = new int [2];
+        boolean visited[] = new boolean[N+1];
+        ArrayList<int[]>[] adj = new ArrayList[N+1]; // 버스 번호가 1부터 시작
+        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> Integer.compare(a[1], b[1]));
+
+        for (int i = 0; i <= N; ++i) 
+        {
+            dp[i] = Integer.MAX_VALUE;
+            adj[i] = new ArrayList<>();
+        }
+
+        int from, to, weight;
+
+        for (int i = 0; i < M; ++i) 
+        {
+            st = new StringTokenizer(br.readLine());
+            from = Integer.parseInt(st.nextToken());
+            to = Integer.parseInt(st.nextToken());
+            weight = Integer.parseInt(st.nextToken());
+            adj[from].add(new int[]{to, weight});
+        }
+
+        st = new StringTokenizer(br.readLine());
+        from = Integer.parseInt(st.nextToken());
+        to = Integer.parseInt(st.nextToken());
+
+        dp[from] = 0;//자기 자신으로 가는건 0
+        pq.offer(new int[]{from, 0});
+
+        while (!pq.isEmpty())
+        {
+            tmp = pq.poll();
+            if(visited[tmp[0]]) continue;
+
+            visited[tmp[0]] = true;
+            for(final int[] iter : adj[tmp[0]])
+            {
+                if(dp[iter[0]] > dp[tmp[0]] + iter[1])
+                {
+                    dp[iter[0]] = dp[tmp[0]] + iter[1];
+                    pq.offer(new int[]{iter[0], dp[iter[0]]});
+                }
+            }
+        }
```

Comment:
이것도 취향 차이이긴 한데 제안 드리고 싶은 방식이 있습니다!
```suggestion
        while (!pq.isEmpty())
        {
            tmp = pq.poll();
            int number = tmp[0];
            if(visited[number]) continue;

            visited[number] = true;
            for(final int[] iter : adj[number])
            {
                int next_number = iter[0], next_cost = iter[1];
                if(dp[next_number] > dp[number] + next_cost)
                {
                    dp[next_number] = dp[number] + next_cost;
                    pq.offer(new int[]{next_number, dp[next_number]});
                }
            }
        }
```
`iter[0]`, `iter[1]`, `tmp[0]` 처럼 의미가 잘 드러나지 않는 변수는 의미가 잘 드러나는 새로운 변수에 저장하면 어떨까요?

현재는 코드 길이가 그렇게 길지 않고, 로직이 엄청 복잡한 게 아니라서 장점이 잘 드러나지는 않지만... 
이미 알고 계셨을 수도 있지만 막 생각나서 부랴부랴 적어봤습니다!

Reply:
그러네요!!  더 명확해서 보기 좋네요 !!! 늘 꼼꼼히 봐줘서 도움이 많이 됩니다. 감사합니다 🍀